### PR TITLE
28739 postman wired mock

### PR DESCRIPTION
**Pull request by victoralfaro-dotcms**
Adding WireMock support to be consumed by dotCMS instance at postman tests.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/ai/api/CompletionsAPI.java [hunk 1]
```diff
@@ -42,20 +42,20 @@ static CompletionsAPI impl() {
...
     /**
-     * this method takes a prompt and returns the AI response based upon that prompt
+     * this method takes a prompt in the form of json and returns a json AI response based upon that prompt
      *
-     * @param promptForm
+     * @param promptJSON
      * @return
      */
-    JSONObject raw(CompletionsForm promptForm);
+    JSONObject raw(JSONObject promptJSON);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/ai/api/CompletionsAPI.java:50]: why using jsonobject, should use Map instead

Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/ai/api/CompletionsAPI.java:50]: @jdotcms JSONObject is used all over the place when it comes to AI API.

File: dotCMS/src/main/java/com/dotcms/ai/AiKeys.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    public static final String MATCHES = "matches";
+    public static final String QUERY = "query";
+    public static final String TOTAL = "total";
+    public static final String THRESHOLD = "threshold";
+    public static final String DOT_CMS_RESULTS = "dotCMSResults";
+    public static final String OPERATOR = "operator";
+    public static final String OFFSET = "offset";
+    public static final String LIMIT = "limit";
+    public static final String COUNT = "count";
+    public static final String INPUT = "input";
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/ai/AiKeys.java:62]: This is all good work @victoralfaro-dotcms but let's send this type of changes in different PRs so it's clear what the actual required changes were.

File: dotCMS/src/main/java/com/dotcms/ai/app/AppConfig.java [hunk 3]
```diff
@@ -11,8 +11,14 @@
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
+/**
+ * The AppConfig class provides a configuration for the AI application.
+ * It includes methods for retrieving configuration values based on given keys.
+ */
 public class AppConfig implements Serializable {
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/ai/app/AppConfig.java:18]: Same here. This could've been a separate PR for including missing javadoc.

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/ai/AiKeys.java (hunk 2 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/ai/AiKeys.java]: Extracted this into a commit and removed from current branch.

File: dotCMS/src/main/java/com/dotcms/ai/app/AppConfig.java (hunk 3 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/ai/app/AppConfig.java]: Extracted this into a commit and removed from current branch.

**Review by jdotcms — APPROVED** [review]
